infrastructure/smart-contracts/contracts/mock/MockRollup.sol

```diff
@@ -0,0 +1,143 @@
+
+    struct GlobalState {
+        bytes32 bytes32Vals;               // Example field for global state; adjust as needed for real use cases
+        uint64[2] u64Vals;                 // Example array for storing two 64-bit unsigned integer values
+    }
+
+    struct Assertion {
+        GlobalState beforeState;           // Represents the global state before a transition
+        GlobalState afterState;            // Represents the global state after a transition
+        uint64 numBlocks;                  // Number of blocks involved in this assertion
+    }
+
+    // Events to log node creation and confirmation actions
+    event NodeCreated(
+        uint64 indexed nodeNum,
+        bytes32 indexed parentNodeHash,
+        bytes32 indexed nodeHash,
+        bytes32 executionHash,
+        Assertion assertion,
+        bytes32 afterInboxBatchAcc,
+        bytes32 wasmModuleRoot,
+        uint256 inboxMaxCount
+    );
+
+    event NodeConfirmed(uint64 indexed nodeNum, bytes32 blockHash, bytes32 sendRoot);
+
+    /**
+     * @notice Creates a new node with provided block hash and send root.
+     * @param blockHash The simulated block hash to associate with this node.
+     * @param sendRoot The simulated send root to associate with this node.
+     */
+    function createNode(
+        uint64 nodeNum,
+        bytes32 blockHash,
+        bytes32 sendRoot
+    ) external {
+        // Input validation to ensure non-zero values are provided
+        require(blockHash != bytes32(0), "Block hash cannot be zero");
+        require(sendRoot != bytes32(0), "Send root cannot be zero");
+
+        // Generating various hashes to simulate node creation
+        bytes32 parentNodeHash = keccak256(abi.encodePacked(blockhash(block.number - 1), block.timestamp, msg.sender));
+        bytes32 nodeHash = keccak256(abi.encodePacked(blockhash(block.number - 2), block.timestamp, msg.sender));
+        bytes32 executionHash = keccak256(abi.encodePacked(blockhash(block.number - 3), block.timestamp, msg.sender));
+        bytes32 afterInboxBatchAcc = keccak256(abi.encodePacked(blockhash(block.number - 4), block.timestamp, msg.sender));
+        bytes32 wasmModuleRoot = keccak256(abi.encodePacked(blockhash(block.number - 5), block.timestamp, msg.sender));
+        
+        // Compute the simulated confirmData hash using the provided blockHash and sendRoot
+        bytes32 confirmData = keccak256(abi.encodePacked(blockHash, sendRoot));
+        
+        // Initialize an assertion structure with dummy data this data will not be used in this mock contract
+        Assertion memory assertion = Assertion(
+            GlobalState({bytes32Vals: bytes32(0), u64Vals: [uint64(0), uint64(0)]}),
+            GlobalState({bytes32Vals: bytes32(0), u64Vals: [uint64(0), uint64(0)]}),
+            uint64(0)
+        );
+
+        // Store the newly created node in the mapping with all its attributes
+        _nodes[latestNodeCreated] = Node({
```

Comment:
This seems wrong to me, why are we mapping the new node we create to the latestNodeCreated counter ?
Shoud this not be `_nodes[nodeNum]` or am I missing something here ?

Reply:
The latestNodeCreated variable was supposed to have been set to NodeNum, but seems it wasn't getting set until after.

Great catch, this has been corrected to use nodeNum and re-deployed.